Dockerfile

```diff
@@ -0,0 +1,9 @@
+# Dockerfile (For Development)
+FROM node:18-alpine
+WORKDIR /app
+COPY package*.json ./
+RUN npm install
+COPY . .
+RUN npm run build            
+EXPOSE 3000
+CMD ["npm", "start"]
```

Comment:
Will this start the project in dev?

I think we want it to run similar to prod.

Reply:
@LeoQiu981112 , I am worried how that might affect hot reloading, if it's a challenge, let me know.

Reply:
If we just run in dev mode, hot reloading works, but if we want to make use of pre-rendered pages and run similar to prod env using `npm run startProduction` , hot-reload doesn't work, also, the below changes were needed to make  prod build work in docker

1. needed to comment out the below line in next.config.js
output: 'export'

2. needed to change basepath to empty otherwise would get 404 for pages in next.config.js
  basePath: process.env.NODE_ENV === 'production' ? '' : '',

sidenote: if we don't comment out `output: 'export'`  when starting prod server, would get below error instead
Error: "next start" does not work with "output: export" configuration. "npx serve@latest out" instead.

and when the app is ran in docker using corresponding command below, it would just show a barebone site and not verydevelopment friendly
CMD ["npx", "serve", "-s", "out", "-l", "3000"]

Reply:
Thanks @LeoQiu981112 ,

1) How does it work right now in prod? It should build the app and npm run startProduction?
2) How does it work in prod?

Why do you need to run this command?

CMD ["npx", "serve", "-s", "out", "-l", "3000"]

Reply:
The priority for this ticket is

1) Mimic prod as much as possible since the build process heavily optimizes the app so local development cannot be trusted
2) Ensure developer ergonomics. (e.g. Hot Reload)

So if we can't satisfy number 2, that might be a sacrifice we need to make.